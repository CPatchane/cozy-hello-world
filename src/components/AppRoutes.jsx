import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'

const Hello = (name) => (<p style='margin-left: 2em'>Just... Hello world! This is {name}</p>)

const Hello1 = () => Hello('View 1')
const Hello2 = () => Hello('View 2')
const Hello3 = () => Hello('View 3')

const AppRoutes = function () {
  return <Switch>
    <Route path='/viewhello1' component={Hello1} />
    <Route path='/viewhello2' component={Hello2} />
    <Route path='/viewhello3' component={Hello3} />
    <Redirect from='/' to='/viewhello1' />
  </Switch>
}

export default AppRoutes
