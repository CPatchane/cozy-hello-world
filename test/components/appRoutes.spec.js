'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import AppRoutes from '../../src/components/AppRoutes'

describe('AppRoutes component', () => {
  it('should be rendered correctly', () => {
    const component = shallow(
      <AppRoutes />
    ).node
    expect(component).toMatchSnapshot()
  })
})
