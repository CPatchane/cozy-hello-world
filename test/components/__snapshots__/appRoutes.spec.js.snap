// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRoutes component should be rendered correctly 1`] = `
<Switch>
  <Route
    component={[Function]}
    path="/viewhello1"
  />
  <Route
    component={[Function]}
    path="/viewhello2"
  />
  <Route
    component={[Function]}
    path="/viewhello3"
  />
  <Redirect
    from="/"
    push={false}
    to="/viewhello1"
  />
</Switch>
`;
